apps/arweave/src/ar_data_sync.erl

```diff
@@ -2931,20 +2949,50 @@ deregister_currently_processed_disk_pool_key(Key, State) ->
 	Keys2 = sets:del_element(Key, Keys),
 	State#sync_data_state{ currently_processed_disk_pool_keys = Keys2 }.
 
+get_merkle_rebase_threshold() ->
+	case ets:lookup(node_state, merkle_rebase_support_threshold) of
+		[] ->
+			infinity;
+		[{_, Threshold}] ->
+			Threshold
+	end.
+
 process_disk_pool_chunk_offset(Iterator, TXRoot, TXPath, AbsoluteOffset, MayConclude, Args,
 		State) ->
 	#sync_data_state{ disk_pool_threshold = DiskPoolThreshold } = State,
-	{Offset, _, _, DataRoot, DataPathHash, _, _, PassedStrictValidation} = Args,
-	case AbsoluteOffset =< ?STRICT_DATA_SPLIT_THRESHOLD orelse PassedStrictValidation of
+	{Offset, _, _, DataRoot, DataPathHash, _, _,
+			PassedBase, PassedStrictValidation, PassedRebaseValidation} = Args,
+	PassedValidation =
+		case {AbsoluteOffset >= get_merkle_rebase_threshold(),
```

Comment:
Is there some way to document this logic? I have sample comment below... but I've been tryin to think of how to change the code so that a comment wouldn't be needed... but coming up empty. Here's a potentially comment:

```
%% Validation passes under the following conditions:
%% 1. If offset is past the rebase_threshold it must pass rebase validation
%% 2. If offset is between the data_split_threshold and rebase_threshold it must pass all 3 validations (base, strict, and rrebase) 
%% 3. If offset is before the data_split_threshold it must either pass all 3 validations, or *just* pass the base validation (and fail the other 2)
```

Reply:
One minor change (not sure if it makes it easier to follow though) - basically if all checks pass, then the overall validation passes regardless of what the Offset is. (I think this is logically consistent with the current code)

```
	PassedValidation =
		case {AbsoluteOffset >= get_merkle_rebase_threshold(),
				AbsoluteOffset >= ?STRICT_DATA_SPLIT_THRESHOLD,
				PassedBase, PassedStrictValidation, PassedRebaseValidation} of
			{_, _, true, true, true} ->
				true;
                         {true, true, _, _, true} ->
				true;
			{false, false, true, false, false} ->
				true;
			_ ->
				false
		end,
```